src/Transformer.ts

```diff
@@ -0,0 +1,400 @@
+import * as path from "path";
+import * as ts from "typescript";
+
+import {CompilerOptions} from "./CompilerOptions";
+import {DecoratorKind} from "./Decorator";
+import {TSTLErrors} from "./Errors";
+import {TransformHelper as transformHelper} from "./TransformHelper";
+import {TSHelper as tsHelper} from "./TSHelper";
+
+export class LuaTransformer {
+    private checker: ts.TypeChecker;
+    private options: CompilerOptions;
+    private context: ts.TransformationContext;
+
+    constructor(checker: ts.TypeChecker, options: CompilerOptions) {
+        this.checker = checker;
+        this.options = options;
+    }
+
+    public transform(node: ts.SourceFile): ts.SourceFile {
+        return ts
+            .transform(node,
+                       [(ctx: ts.TransformationContext) => {
+                           this.context = ctx;
+                           return this.transformSourceFile.bind(this);
+                       }],
+                       this.options)
+            .transformed[0];
+    }
+
+    public visitEachChild<T extends ts.Node>(node: T): T {
+        return ts.visitEachChild(node, child => this.visitor(child), this.context);
+    }
+
+    public transformSourceFile(node: ts.SourceFile): ts.SourceFile {
+        if (node.isDeclarationFile) {
+            return node;
+        }
+        return this.visitEachChild(node);
+    }
+
+    public visitor(node: ts.Node): ts.VisitResult<ts.Node> {
+        // Ignore declarations
+        if (node.modifiers && node.modifiers.some(modifier => modifier.kind === ts.SyntaxKind.DeclareKeyword)) {
+            return undefined;
+        }
+        switch (node.kind) {
+            // Declarations
+            case ts.SyntaxKind.ImportDeclaration:
+                return this.visitImportDeclaration(node as ts.ImportDeclaration);
```

Comment:
Random idea: Maybe we should have default `this.ignoreNode(node)` (returns undefined) and `this.visitDefault(node)` (returns node)?

Reply:
Ideally all those method stubs should contain an actual implementation in the future. If we figure out that we don't need to transform a node (ignore it) we can just delete the case from the switch statement.